src/util/metadata/side_metadata/helpers.rs

```diff
@@ -197,15 +228,21 @@ pub fn find_last_non_zero_bit_in_metadata_bytes(
     let mut mapped_chunk = Address::MAX;
     while cur > meta_start {
         // If we can check the whole word, set step to word size. Otherwise, the step is 1 (byte) and we check byte.
-        let step = if cur.is_aligned_to(BYTES_IN_ADDRESS)
-            && cur.align_down(BYTES_IN_ADDRESS) >= meta_start
-        {
+        let step = if cur.is_aligned_to(BYTES_IN_ADDRESS) && cur - BYTES_IN_ADDRESS >= meta_start {
```

Comment:
This is the other fix that is needed. The old code may check an address that is below `meta_start`.